homeassistant/components/zabbix/__init__.py

```diff
@@ -46,13 +90,208 @@ def setup(hass, config):
     username = conf.get(CONF_USERNAME)
     password = conf.get(CONF_PASSWORD)
 
-    zapi = ZabbixAPI(url)
+    publish_states_host = conf.get(CONF_PUBLISH_STATES_HOST)
+
+    include = conf.get(CONF_INCLUDE, {})
+    exclude = conf.get(CONF_EXCLUDE, {})
+    whitelist_e = set(include.get(CONF_ENTITIES, []))
+    whitelist_d = set(include.get(CONF_DOMAINS, []))
+    blacklist_e = set(exclude.get(CONF_ENTITIES, []))
+    blacklist_d = set(exclude.get(CONF_DOMAINS, []))
+
+    max_tries = conf.get(CONF_RETRY_COUNT)
+
     try:
-        zapi.login(username, password)
+        zapi = ZabbixAPI(url=url, user=username, password=password)
         _LOGGER.info("Connected to Zabbix API Version %s", zapi.api_version())
     except ZabbixAPIException as login_exception:
         _LOGGER.error("Unable to login to the Zabbix API: %s", login_exception)
         return False
+    except HTTPError as http_error:
+        _LOGGER.error("HTTPError when connecting to Zabbix API: %s", http_error)
+        zapi = None
+        # Continue even though Zabbix API couldn't be initialized. Zabbix sender later may still work.
 
     hass.data[DOMAIN] = zapi
+
+    def event_to_metrics(event, float_keys, string_keys):
+        """Add an event to the outgoing Zabbix list."""
+        state = event.data.get("new_state")
+        if (
+            state is None
+            or state.state in (STATE_UNKNOWN, "", STATE_UNAVAILABLE)
+            or state.entity_id in blacklist_e
+            or state.domain in blacklist_d
+        ):
+            return
+
+        entity_id = state.entity_id
+        floats = {}
+        strings = {}
+        try:
+            if (
+                (whitelist_e or whitelist_d)
+                and entity_id not in whitelist_e
+                and state.domain not in whitelist_d
+            ):
+                return
+
+            _state_as_value = float(state.state)
+            floats[entity_id] = _state_as_value
+        except ValueError:
+            try:
+                _state_as_value = float(state_helper.state_as_number(state))
+                floats[entity_id] = _state_as_value
+            except ValueError:
+                strings[entity_id] = state.state
+
+        for key, value in state.attributes.items():
+            # For each value we try to cast it as float
+            # But if we can not do it we store the value
+            # as string
+            attribute_id = f"{entity_id}/{key}"
+            try:
+                float_value = float(value)
+                if math.isfinite(float_value):
+                    floats[attribute_id] = float_value
+                else:
+                    strings[attribute_id] = str(value)
+            except (ValueError, TypeError):
+                strings[attribute_id] = str(value)
+
+        metrics = []
+        float_keys_count = len(float_keys)
+        float_keys.update(floats.keys())
```

Comment:
The keys will be used by default.
```suggestion
        float_keys.update(floats)
```

Reply:
Ok